semgrep/semgrep/output.py

```diff
@@ -171,26 +173,27 @@ def build_timing_summary(
     GREEN_COLOR = colorama.Fore.GREEN if color_output else ""
     YELLOW_COLOR = colorama.Fore.YELLOW if color_output else ""
 
-    rule_parsing_time = 0.0
-    file_parsing_time = 0.0
-    rule_timings = {}
-    file_timings: Dict[str, float] = {}
-    for rule in rules:
-        parse_time = profiling_data.get_parse_time(rule.id)
-        rule_timings[rule.id] = (parse_time, 0.0)
-        rule_parsing_time += parse_time
-        for target in targets:
-            path_str = get_path_str(target)
-            times = profiling_data.get_run_times(rule.id, path_str)
-            rule_timings[rule.id] = tuple(
-                x + y
-                for x, y in zip(
-                    rule_timings[rule.id],
-                    (times.run_time - times.parse_time, times.match_time),
-                )
-            )  # type: ignore
-            file_timings[path_str] = file_timings.get(path_str, 0.0) + times.run_time
-            file_parsing_time += times.parse_time
+    time_data = _build_time_json(rules, targets, profiling_data, profiler)
```

Comment:
Oh, since we're getting the profiler now, you can also output the total wall time